### PR TITLE
Remove str_join.h

### PR DESCRIPTION
**Pull request by desertfire**

### CODE REVIEW

File: lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h [hunk 1]
```diff
@@ -30,6 +30,7 @@ class Scalar : public TsNode {
 
 torch::lazy::hash_t ScalarHash(const at::Scalar& s);
 
+// TODO: remove this after the aten version is exposed properly
```
Comment (desertfire) [inline lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h:33]: I was having a hard time to access the same function from `#include <ATen/core/Formatting.h>`. We might need to change pytorch core to access it.

Comment (wconstab) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h:33]: huh.  namespace issue?

File: tools/codegen/gen_lazy_tensor.py [hunk 2]
```diff
@@ -197,10 +197,10 @@ def concat_map_codegen(func: Callable[[NativeFunction], Sequence[str]],
                 "vector",
             ]],
             'lazy_ir_inc': [f'#include "{path}"' for path in [
+                "torch/csrc/lazy/core/hash.h",
                 "torch/csrc/lazy/core/ir.h",
                 "lazy_tensor_core/csrc/ts_backend/ts_node_lowering.h",
-                "torch/csrc/lazy/core/hash.h",
-                "lazy_tensors/str_join.h",
+                "lazy_tensor_core/csrc/ops/scalar.h",
```
Comment (wconstab) [inline tools/codegen/gen_lazy_tensor.py:203]: hah. ok,  go for it:)

**Review by wconstab — APPROVED** [review]

File: lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h (hunk 1 above)
Comment (alanwaketan) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h:33]: Why remove it?

**Review by alanwaketan — APPROVED** [review]
LGTM.

File: lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h (hunk 1 above)
Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h]: > Why remove it?

I think it is a temp workaround from Alex. We could just use the one from https://github.com/pytorch/pytorch/blob/f5daa9f76bcf31bab61feefc668de523bc667c2f/aten/src/ATen/core/Formatting.h#L23

Comment (desertfire) [reply lazy_tensor_core/lazy_tensor_core/csrc/ops/scalar.h]: > huh. namespace issue?

Yes. I have uploaded a new version which works now after adding `using namespace at;`. I initially put that in scalar.cpp so it did not work...

**Pull request merged**